client/landing/stepper/README.md

```diff
@@ -1,106 +1,387 @@
+## Useful utilities and hooks
+
+Please check out the [hooks](/client/landing/stepper/hooks) and [utils](/client/landing/stepper/utils) folders. They have many useful utilities that make building flows easier.
 
-## State management
+## Troubleshooting
 
-Steps shouldn't have any props other than the `navigation` prop which contains the return value of `useStepNavigation`. This object (`navigation`) allows the step to `submit` when done or to move to other steps or to `goNext`, `goBack`, etc..
+### TypeScript is complaining about the type of `useStepProps`
 
-The rationale behind this is the following:
+The `useStepProps` hooks allows you to pass props to your steps. TS will only allow your return value to match the steps that exist in your flow and their props. And the return value does not match, it will complain about the whole function, not a single slug or a prop. To debug, it's best to remove the props of all the steps and add them one by one until it becomes red again.
 
-1. **Small API**: when steps have such small API surface area, they can be reused easily.
-2. **Centralized state**: the `useStepNavigation` hook makes decisions based on the entire flow's state, so the state has to be centralized, i.e: lives entirely in the `onboard` store (or any store inside `packages/data-stores`).
+```ts
+const flow = {
+	//👇 the TS red line will be here 
+	useStepsProps() {
+		return {
+			domains: {
+				allowFree: true,
+			},
+			//👇 not here
+			nonExistentStep: {				
+				title: 'this step does not exist',
+			},
+		};
+	},
+};
+```
 
-All the state should be either in the onboarding store or the internal state of the step. This creates a flat data tree that is exactly one level deep and allows steps to be easily created and re-used.
+### TypeScript is complaining about the arguments I'm passing to `submit`
 
-It takes five minutes to add a new field to the `onboard` store. Such as [`intent`](../../../packages/data-stores/src/onboard/reducer.ts) or [`storeType`](../../../packages/data-stores/src/onboard/reducer.ts).
+Sometimes, in your step, you call `submit({ answer: 'yes' })` and TS will complain about the argument. This means you haven't typed your step correctly.
 
-## Preferring packages
+In this example, the step's type should look like so
 
-When creating new code, it's highly-encouraged to put it into packages, whether existing like `onboarding` or new packages that you create. This allows us to create flows in the future that live in wp-admin or wherever.
+```ts
+const YourStep: Step< {
+	submits: {
+		answer: 'yes' | 'no';
+	};
+} >;
+```
 
 ## Help and feedback
 
-Please feel free to reach out to Team Vertex for any feedback or if you need help.
+Please feel free to reach out to Team T-Rex, or post in `#dotcom-stepper` for any feedback or if you need help.
```

Comment:
Same note as above. Should team Calypso be responsible for Stepper?

Reply:
The best team/people would be those who build and maintain flows. Team Calypso seldom do that.

Reply:
Got it, then I guess it's fine to keep T-Rex as generic maintainer, but we should also mention to ping owner teams when it comes to individual flows, based on the flow's readme.

Reply:
Yes 100%.